modules/wayfire/hm.nix

```diff
@@ -0,0 +1,63 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+{
+  options.stylix.targets.wayfire.enable =
+    config.lib.stylix.mkEnableTarget "wayfire" true;
+
+  config =
+    lib.mkIf (config.stylix.enable && config.stylix.targets.wayfire.enable)
+      (
+        let
+          inherit (config.lib.stylix) colors;
+          rgba = rgb: a: "\\#${rgb}${a}";
+          rgb = (lib.flip rgba) "ff";
+
+          wayfireConfig = config.wayland.windowManager.wayfire;
+
+          wayfireBackground = pkgs.runCommand "wayfire-background.png" { } ''
+            ${lib.getExe' pkgs.imagemagick "convert"} ${config.stylix.image} $out
+          '';
```

Comment:
> i dont understand what you mean by this. could you please elaborate what problem you're seeing here?

Essentially, does this work even when `stylix.image` is not a PNG file? Note that the file extension does not dictate the actual file type:

```console
$ file image.png
image.png: PNG image data, [...]

$ file image.jpg
image.jpg: PNG image data, [...]
```

Just try it with some files and tell me if it worked.

> edit: didnt intend to sound that angry, sorry

No worries. My explanation was not very clear :)

Reply:
I believe the `convert` command reads the file extensions to determine what to convert from/to. There may be an option to set this explicitly.

Reply:
Let's just merge it as-is for now.